Migrate website to Jekyll

justinjereza opened the pull request:
README.md created to guide contributors.

justinjereza committed:
Migrated website to Jekyll.

justinjereza committed:
Override minima default _includes/head.html.

Minima uses jekyll-seo-tag and by default appends the description
to the title. It makes the title look too lengthy.

justinjereza committed:
Added vim swap files to gitignore.

zakame commented on Containerfile.build:
```diff
@@ -0,0 +1,21 @@
+FROM docker.io/library/alpine:latest
+
+ARG RUBY="ruby-3.0.1"
+ARG RUBY_URL="https://cache.ruby-lang.org/pub/ruby/3.0/${RUBY}.tar.gz"
+
+RUN apk --update-cache upgrade
+RUN apk add build-base tar zlib-dev openssl-dev
+ADD [ "${RUBY_URL}", "/tmp/" ]
+
+RUN tar xf /tmp/${RUBY}.tar.gz -C /tmp
+WORKDIR /tmp/${RUBY}
+RUN ./configure
+RUN make install
```
Maybe this can all be simply `FROM docker.io/library/ruby:3.0-alpine` instead?

justinjereza replied:
That is true. For some reason, I didn't find that on Docker Hub until you mentioned it. Given the changes, I'll have to make a new pull request to just roll everything up.

justinjereza replied:
All Containerfiles are now based on `docker.io/library/ruby:alpine`. I forgot that GitHub automatically adds new commits to a pull request so no need to make a new one. :sweat_smile:

zakame commented on Containerfile.build:
```diff
@@ -0,0 +1,21 @@
+WORKDIR /root
+RUN rm -r /tmp/${RUBY} /tmp/${RUBY}.tar.gz /var/cache/apk/* /usr/local/lib/ruby/gems/3.0.0/cache
+
+ENTRYPOINT [ "/usr/bin/tar" ]
+CMD [ "czP", "/usr/local" ]
```
No real need for `ENTRYPOINT` here since no off-container paths are imported (e.g. as parameters passed to `docker run`.)

```suggestion
CMD [ "/usr/bin/tar", "czP", "/usr/local" ]
```

justinjereza replied:
Yeah, it's there's not much of a use case for changing just the parameters of tar anyway. This will be eliminated anyway by basing everything on `docker.io/library/ruby:alpine`.

justinjereza commented on Containerfile.build:
```diff
@@ -0,0 +1,21 @@
+FROM docker.io/library/alpine:latest
```
I forgot that one reason I made a separate container for building is to keep the compiler out of the final image. Some gems require a compiler. We'll have to retain the build container. We can base it on `docker.io/library/ruby:alpine` though. The final image will still be based on `docker.io/library/alpine:latest` because of this.

justinjereza replied:
In the end, both Containerfiles were rebased on `docker.io/library/alpine:latest`.

justinjereza committed:
Clarified that building the image should be done inside the repository directory.

justinjereza committed:
Explicitly added --file Containerfile in image building example.

This guarantees that `alias podman="docker"` will work.

justinjereza committed:
Rebased Containerfiles to `docker.io/library/ruby:alpine`.

Removed ENTRYPOINT from Containerfile.build. There is no reason to
treat the resulting image as an application.

zakame commented on Gemfile:
```diff
@@ -0,0 +1,32 @@
+  gem "jekyll-feed", "~> 0.12"
+end
+
+# Windows and JRuby does not include zoneinfo files, so bundle the tzinfo-data gem
+# and associated library.
+platforms :mingw, :x64_mingw, :mswin, :jruby do
+  gem "tzinfo", "~> 1.2"
+  gem "tzinfo-data"
+end
+
+# Performance-booster for watching directories on Windows
+gem "wdm", "~> 0.1.1", :platforms => [:mingw, :x64_mingw, :mswin]
+
```
Can probably cut these out given its running under a Linux container - not too urgent though :bow:

justinjereza replied:
Since they only get installed on Windows, I think they're ok to leave in. If for some reason somebody decides to make changes on Windows, they get an option. Harmless when running it on Linux since they don't get installed anyway.

zakame commented on Gemfile.lock:
```diff
@@ -0,0 +1,82 @@
+  jekyll (~> 4.2.0)
+  jekyll-feed (~> 0.12)
+  minima (~> 2.5)
+  tzinfo (~> 1.2)
+  tzinfo-data
+  wdm (~> 0.1.1)
```
ditto

zakame reviewed (approved):
Thanks @justinjereza, LGTM! :tada:

ericpareja reviewed (approved):
lgtm
